examples/connext_dds/lbediscovery_xml_app_creation/c++11/ShapeType_subscriber.cxx

```diff
@@ -58,21 +58,26 @@ void run_subscriber_application(
                     "ShapeTypeSubscriberParticipant",
                     params);
 
+    const unsigned int reader_number = 4;
     const std::string reader_names[] = { "Sub::RED_DR",
                                          "Sub::RED_DR#1",
                                          "Sub::BLUE_DR",
                                          "Sub::BLUE_DR#1" };
     std::vector<dds::sub::DataReader<ShapeType>> readers;
-    std::vector<dds::core::cond::WaitSet> waitsets(4);
-    std::vector<unsigned int> samples_read(4);
+    std::vector<dds::core::cond::WaitSet> waitsets(reader_number);
+    std::vector<unsigned int> samples_read(reader_number);
+    dds::sub::DataReader<ShapeType> reader = dds::core::null;
 
     // Retrieve the four DataReaders and configure their
     // ReadConditions and Waitsets
-    for (int i = 0; i < 4; i++) {
-        readers.push_back(rti::sub::find_datareader_by_name<
-                          dds::sub::DataReader<ShapeType>>(
-                participant,
-                reader_names[i]));
+    for (int i = 0; i < reader_number; i++) {
+        reader = rti::sub::find_datareader_by_name<
```

Comment:
declare the variable `reader` here as `auto` instead of initializing it to `null` above.